p2p/peer_tracker.go

```diff
@@ -24,9 +25,21 @@ var (
 	// otherwise it will be removed on the next GC cycle.
 	maxAwaitingTime = time.Hour
 	// gcCycle defines the duration after which the peerTracker starts removing peers.
-	gcCycle = time.Minute * 30
+	gcCycleDefault = time.Minute * 30
 )
 
+type params struct {
+	gcCycle time.Duration
+}
+
+type PTOption func(*params)
```

Comment:
I don't think there's enough reason to warrant this being an opt right now - let's just decrease the default to 5 min GC cycle. 

It makes the package quite messy to have different option types as well.

Reply:
Indeed, that's a relic from a broken testing methodology